File: oeml-sdk/javascript-closure-angular/API/Client/ExecutionReport.js
goog.provide('API.Client.ExecutionReport');

/**
 * @record
 */
API.Client.ExecutionReport = function() {}

/**
 * Result type name
 * @type {!string}
 * @export
 */
API.Client.ExecutionReport.prototype.type;

/**
 * Exchange name
 * @type {!string}
 * @export
 */
API.Client.ExecutionReport.prototype.exchangeId;

/**
 * Client unique identifier for the trade.
 * @type {!string}
 * @export
 */
API.Client.ExecutionReport.prototype.id;

/**
 * Hash client id
 * @type {!string}
 * @export
 */
API.Client.ExecutionReport.prototype.clientOrderIdFormatExchange;

/**
 * Exchange order id
 * @type {!string}
 * @export
 */
API.Client.ExecutionReport.prototype.exchangeOrderId;

/**
 * Amount open
 * @type {!number}
 * @export
 */
API.Client.ExecutionReport.prototype.amountOpen;

/**
 * Amount filled
 * @type {!number}
 * @export
 */
API.Client.ExecutionReport.prototype.amountFilled;

/**
 * @type {!API.Client.OrdStatus}
 * @export
 */
API.Client.ExecutionReport.prototype.status;

/**
 * History of order status changes
 * @type {!Array<!Array<!string>>}
 * @export
 */
API.Client.ExecutionReport.prototype.timeOrder;

/**
 * Error message
 * @type {!string}
 * @export
 */
API.Client.ExecutionReport.prototype.errorMessage;

/**
 * Client unique identifier for the trade.
 * @type {!string}
 * @export
 */
API.Client.ExecutionReport.prototype.clientOrderId;

/**
 * The symbol of the order.
 * @type {!string}
 * @export
 */
API.Client.ExecutionReport.prototype.symbolExchange;

/**
 * The CoinAPI symbol of the order.
 * @type {!string}
 * @export
 */
API.Client.ExecutionReport.prototype.symbolCoinapi;

/**
 * Quoted decimal amount to purchase.
 * @type {!number}
 * @export
 */
API.Client.ExecutionReport.prototype.amountOrder;

/**
 * Quoted decimal amount to spend per unit.
 * @type {!number}
 * @export
 */
API.Client.ExecutionReport.prototype.price;

/**
 * Buy or Sell
 * @type {!string}
 * @export
 */
API.Client.ExecutionReport.prototype.side;

/**
 * The order type.
 * @type {!string}
 * @export
 */
API.Client.ExecutionReport.prototype.orderType;

/**
 * @type {!API.Client.TimeInForce}
 * @export
 */
API.Client.ExecutionReport.prototype.timeInForce;

/**
 * Required for orders with time_in_force = GOOD_TILL_TIME_EXCHANGE, GOOD_TILL_TIME_OMS
 * @type {!API.Client.date}
 * @export
 */
API.Client.ExecutionReport.prototype.expireTime;

/**
 * Order execution instructions are documented in the separate section: <a href=\"#oeml-order-params-exec\">OEML / Starter Guide / Order parameters / Execution instructions</a>
 * @type {!Array<!string>}
 * @export
 */
API.Client.ExecutionReport.prototype.execInst;

/** @enum {string} */
API.Client.ExecutionReport.SideEnum = { 
  BUY: 'BUY',
  SELL: 'SELL',
}
/** @enum {string} */
API.Client.ExecutionReport.OrderTypeEnum = { 
  LIMIT: 'LIMIT',
}
/** @enum {string} */
API.Client.ExecutionReport.Array&lt;!ExecInstEnum&gt; = { 
  MAKER_OR_CANCEL: 'MAKER_OR_CANCEL',
  AUCTION_ONLY: 'AUCTION_ONLY',
  INDICATION_OF_INTEREST: 'INDICATION_OF_INTEREST',
}

File: oeml-sdk/dart-dio/lib/model/execution_report.dart
            import 'package:openapi/model/ord_status.dart';
            import 'package:built_collection/built_collection.dart';
            import 'package:openapi/model/time_in_force.dart';
        import 'package:built_value/built_value.dart';
import 'package:built_value/serializer.dart';

part 'execution_report.g.dart';

abstract class ExecutionReport implements Built<ExecutionReport, ExecutionReportBuilder> {

    /* Exchange name */
        @nullable
    @BuiltValueField(wireName: r'exchange_id')
    String get exchangeId;
    /* Client unique identifier for the trade. */
        @nullable
    @BuiltValueField(wireName: r'id')
    String get id;
    /* Hash client id */
        @nullable
    @BuiltValueField(wireName: r'client_order_id_format_exchange')
    String get clientOrderIdFormatExchange;
    /* Exchange order id */
        @nullable
    @BuiltValueField(wireName: r'exchange_order_id')
    String get exchangeOrderId;
    /* Amount open */
        @nullable
    @BuiltValueField(wireName: r'amount_open')
    num get amountOpen;
    /* Amount filled */
        @nullable
    @BuiltValueField(wireName: r'amount_filled')
    num get amountFilled;
    
        @nullable
    @BuiltValueField(wireName: r'status')
    OrdStatus get status;
        //enum statusEnum {  RECEIVED,  ROUTING,  ROUTED,  NEW,  PENDING_CANCEL,  PARTIALLY_FILLED,  FILLED,  CANCELED,  REJECTED,  };
    /* History of order status changes */
        @nullable
    @BuiltValueField(wireName: r'time_order')
    BuiltList<BuiltList<String>> get timeOrder;
    /* Error message */
        @nullable
    @BuiltValueField(wireName: r'error_message')
    String get errorMessage;
    /* Client unique identifier for the trade. */
        @nullable
    @BuiltValueField(wireName: r'client_order_id')
    String get clientOrderId;
    /* The symbol of the order. */
        @nullable
    @BuiltValueField(wireName: r'symbol_exchange')
    String get symbolExchange;
    /* The CoinAPI symbol of the order. */
        @nullable
    @BuiltValueField(wireName: r'symbol_coinapi')
    String get symbolCoinapi;
    /* Quoted decimal amount to purchase. */
        @nullable
    @BuiltValueField(wireName: r'amount_order')
    num get amountOrder;
    /* Quoted decimal amount to spend per unit. */
        @nullable
    @BuiltValueField(wireName: r'price')
    num get price;
    /* Buy or Sell */
        @nullable
    @BuiltValueField(wireName: r'side')
    String get side;
        //enum sideEnum {  BUY,  SELL,  };
    /* The order type. */
        @nullable
    @BuiltValueField(wireName: r'order_type')
    String get orderType;
        //enum orderTypeEnum {  LIMIT,  };
    
        @nullable
    @BuiltValueField(wireName: r'time_in_force')
    TimeInForce get timeInForce;
        //enum timeInForceEnum {  GOOD_TILL_CANCEL,  GOOD_TILL_TIME_EXCHANGE,  GOOD_TILL_TIME_OMS,  FILL_OR_KILL,  IMMEDIATE_OR_CANCEL,  };
    /* Required for orders with time_in_force = GOOD_TILL_TIME_EXCHANGE, GOOD_TILL_TIME_OMS */
        @nullable
    @BuiltValueField(wireName: r'expire_time')
    DateTime get expireTime;
    /* Order execution instructions are documented in the separate section: <a href=\"#oeml-order-params-exec\">OEML / Starter Guide / Order parameters / Execution instructions</a> */
        @nullable
    @BuiltValueField(wireName: r'exec_inst')
    BuiltList<String> get execInst;
        //enum execInstEnum {  MAKER_OR_CANCEL,  AUCTION_ONLY,  INDICATION_OF_INTEREST,  };

    // Boilerplate code needed to wire-up generated code
    ExecutionReport._();

    factory ExecutionReport([updates(ExecutionReportBuilder b)]) = _$ExecutionReport;
    static Serializer<ExecutionReport> get serializer => _$executionReportSerializer;

}

